src/index.ts

```diff
@@ -16,7 +16,7 @@ function odata2openapi(metadataUrl: string, options?: Options): Promise<Swagger>
     };
   }
 
-  return get(protocol, host, path).then(parse).then(service => convert(service.entitySets, options, service.version))
+  return get(protocol, host, path, port).then(parse).then(service => convert(service.entitySets, options, service.version))
```

Comment:
The `host` variable seems undefined? You have renamed it to `hostname` during destructuring.

Reply:
Good catch.  Will update.